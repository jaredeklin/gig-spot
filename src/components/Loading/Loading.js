import React from 'react';
import PropTypes from 'prop-types';
import loadingGif from '../../images/spinning-7-inch.gif';

export const Loading = ({ message }) => (
  <div className="loading">
    <img src={loadingGif} className="loading-gif" alt="loading" />
    <h2>{message}</h2>
  </div>
);

Loading.propTypes = {
  message: PropTypes.string
};
