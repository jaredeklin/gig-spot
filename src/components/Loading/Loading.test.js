import React from 'react';
import { shallow } from 'enzyme';
import { Loading } from './Loading';

describe('Loading component', () => {
  it('should match the snapshot', () => {
    const mockMessage = 'Oh Snaps!';
    const wrapper = shallow(<Loading message={mockMessage} />);

    expect(wrapper).toMatchSnapshot();
  });
});
