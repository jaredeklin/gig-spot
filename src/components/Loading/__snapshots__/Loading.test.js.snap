// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    message="Oh Snaps!"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="loading"
          className="loading-gif"
          src="spinning-7-inch.gif"
/>,
        <h2>
          Oh Snaps!
</h2>,
      ],
      "className": "loading",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "loading",
          "className": "loading-gif",
          "src": "spinning-7-inch.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Oh Snaps!",
        },
        "ref": null,
        "rendered": "Oh Snaps!",
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="loading"
            className="loading-gif"
            src="spinning-7-inch.gif"
/>,
          <h2>
            Oh Snaps!
</h2>,
        ],
        "className": "loading",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "loading",
            "className": "loading-gif",
            "src": "spinning-7-inch.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Oh Snaps!",
          },
          "ref": null,
          "rendered": "Oh Snaps!",
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
